### PR TITLE
Mpu6050 part2

### PR DESCRIPTION
**Pull request by Happybolt**
# MPU6050 Practice. Kernel object module development

## Homework

1. Fix the errors in kernel module sources and developed dts.
2. Provide precision 0.001 C for temperature.
3. Measure the performance by getting all values in cycle during 1 second.
4. Improve the performance and measure it.

### CODE REVIEW

**Comment by Happybolt** [comment]
Optimization results are described in commits.
Duplicate:
 1.  Productivity increased by 29.23%
 2.  When reading will be accel_x -> accel_y -> accel_z  productivity increased by 200%.
(regarding commit "MPU650: Optimization")  before optimization
productivity increased by 283%.
 3.   When reading will be 
accel_x->accel_y->accel_z->temp->gyro_x->gyro_y->gyro_z
not optimization: productivity increased by 476.92%
optimization: productivity increased by 647.44%
  When reading will be:
accel->temp->gyro
productivity increased by 219.23

File: mpu6050/testing/MeasurePerformance.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+if [[ $1 = "-h" || $1 = "--help" ]]; then
+	echo "-h, --help     display this help and exit"
+	echo "[time] 	     testing time ms (default 1(s))"
+	exit 0
+fi
+
+if [[ ! -n $1 ]]; then
```
Comment (yekovalyov) [inline mpu6050/testing/MeasurePerformance.sh]: ! -n  is the same like -z
What is the reason for negative condition to if-else block with 2 branches?

File: mpu6050/mpu6050.c [hunk 2]
```diff
@@ -290,7 +341,7 @@ static void mpu6050_exit(void)
 module_init(mpu6050_init);
 module_exit(mpu6050_exit);
 
-MODULE_AUTHOR("Andriy.Khulap <andriy.khulap@globallogic.com>");
```
Comment (yekovalyov) [inline mpu6050/mpu6050.c:293]: You can add  another MODULE_AUTHOR without removing the previous line.

File: mpu6050/mpu6050.c [hunk 3]
```diff
@@ -26,17 +29,23 @@ static int mpu6050_read_data(void)
 		return -ENODEV;
 
 	/* accel */
-	g_mpu6050_data.accel_values[0] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_ACCEL_XOUT_H));
-	g_mpu6050_data.accel_values[1] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_ACCEL_YOUT_H));
-	g_mpu6050_data.accel_values[2] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_ACCEL_ZOUT_H));
+	g_mpu6050_data.accel_values[0] =
+		(s16)(RD_WORD_SW(drv_client, REG_ACCEL_XOUT_H));
```
Comment (yekovalyov) [inline mpu6050/mpu6050.c]: This macros is always used for return signed value. May be it is better to include this conversation in macros.

File: mpu6050/testing/MeasurePerformance.sh (hunk 1 above)
Comment (Happybolt) [reply mpu6050/testing/MeasurePerformance.sh]: To select a variable value (default or custom). 
Can be written as:
DEF_TIMEOUT=1000
TIMEOUT=$ {1: - $ DEF_TIMEOUT}

**Comment by yekovalyov** [comment]
You did good code redesign. The performance changes are significant.

File: mpu6050/testing/MeasurePerformance.sh (hunk 1 above)
Comment (Happybolt) [reply mpu6050/testing/MeasurePerformance.sh]: Replaced by :
TIMEOUT=${1:-1000}

File: mpu6050/mpu6050.c (hunk 2 above)
Comment (Happybolt) [reply mpu6050/mpu6050.c:293]: Only added by "MODULE_AUTHOR".

File: mpu6050/mpu6050.c (hunk 3 above)
Comment (Happybolt) [reply mpu6050/mpu6050.c]: Included in the macro.

**Pull request merged**